### PR TITLE
fix: skip immutable databases in pg_database default-monitoring query

### LINKED ISSUE #4978: [Bug]: frozen xid reported for template databases
### Is there an existing issue already for this bug?

- [X] I have searched for an existing issue, and could not find anything. I believe this is a new bug.

### I have read the troubleshooting guide

- [X] I have read the troubleshooting guide and I think this is a new bug.

### I am running a supported version of CloudNativePG

- [X] I have read the troubleshooting guide and I think this is a new bug.

### Contact Details

smiyc@pm.me

### Version

1.23.2

### What version of Kubernetes are you using?

1.30 (unsupported)

### What is your Kubernetes environment?

Self-managed: kind (evaluation)

### How did you install the operator?

YAML manifest

### What happened?

With the default monitoring and prometheus rules in place the operator is alerting for frozen xid in template0 and template1.

I would exclude template0 and template1 from the metric in
https://github.com/cloudnative-pg/cloudnative-pg/blob/60d48ea6f95cfbae13bc61079b057c0232cfa4fd/config/manager/default-monitoring.yaml#L93

...and rename PGDatabase to PGDatabaseFrozenXid in
https://github.com/cloudnative-pg/cloudnative-pg/blob/57fd494b7bb1ac362094621a37537d6870c8ac76/docs/src/samples/monitoring/prometheusrule.yaml#L30

Slack: https://cloudnativepg.slack.com/archives/C03AX0J5P29/p1719324392606069

### Cluster resource

_No response_

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by smiyc**
This patch excludes immutable databases (databases not allowing connections) from monitoring and alerting. Specifically, the `template0` database is immutable and should not trigger the `PGDatabaseXidAge` alert (formerly known as `PGDatabase`), as all its content is frozen.

Closes #4978

### CODE REVIEW

File: config/manager/default-monitoring.yaml [hunk 1]
```diff
@@ -76,13 +76,14 @@ data:
...
 
-    pg_database:
+    pg_database_frozen_xid:
       query: |
         SELECT datname
           , pg_catalog.pg_database_size(datname) AS size_bytes
           , pg_catalog.age(datfrozenxid) AS xid_age
           , pg_catalog.mxid_age(datminmxid) AS mxid_age
         FROM pg_catalog.pg_database
+          WHERE datname not in ('template0','template1')
```
Comment (mnencia) [inline config/manager/default-monitoring.yaml:86]: I think that you should not exclude `template1` from the check. If it allows connections, it should be checked. I suggest excluding all and only the databases that do not allow connections.
```suggestion
          WHERE datallowconn
```

File: config/manager/default-monitoring.yaml [hunk 2]
```diff
@@ -76,13 +76,14 @@ data:
             usage: "GAUGE"
             description: "Total number of backends that are currently waiting on other queries"
 
-    pg_database:
+    pg_database_frozen_xid:
```
[the remark this answers is not in the document]
Comment (smiyc) [reply config/manager/default-monitoring.yaml:79]: no, but I'd like to rename the alert, if that's ok

File: config/manager/default-monitoring.yaml [hunk 3]
```diff
@@ -83,6 +83,7 @@ data:
           , pg_catalog.age(datfrozenxid) AS xid_age
           , pg_catalog.mxid_age(datminmxid) AS mxid_age
         FROM pg_catalog.pg_database
+          WHERE datallowconn = 't'
```
Comment (mnencia) [inline config/manager/default-monitoring.yaml:86]: Nit picking (indentation and a boolean doesn't need to be compared to true)
```suggestion
        WHERE datallowconn
```

**Review by mnencia — APPROVED** [review]

**Review by jsilvela — APPROVED** [review]

**Pull request merged**